### PR TITLE
implement eth_getTransaction

### PR DESCRIPTION
**Pull request by shunjizhan**
## Change
- implemented `eth_getTransaction` method. Note that I created a new method `getTransactionByHash` instead of using the `getTransaction` from base provider, since there are a lot of type mismatch.
- updated eslint rules to ignore some common warning (in the future we will fix other lint errors and try to make linting pass)

## Test
added some unit tests for the `eth_getTransaction` RPC endpoint

### CODE REVIEW

File: eth-providers/src/base-provider.ts [hunk 1]
```diff
@@ -744,7 +767,7 @@ export abstract class BaseProvider extends AbstractProvider {
 
     // to and contractAddress may be undefined
     return {
-      confirmations: 1,
+      confirmations: DEFAULT_CONFIRMATIONS,
```
Comment (xlc) [inline eth-providers/src/base-provider.ts:770]: should be the confirmation of the block. i.e. latest block number minus query block number

File: eth-providers/src/base-provider.ts [hunk 2]
```diff
@@ -766,15 +789,45 @@ export abstract class BaseProvider extends AbstractProvider {
...
+    const tx = await getTxReceiptByHash(transactionHash);
+
+    if (!tx) {
+      return logger.throwError(`transaction hash not found`, Logger.errors.UNKNOWN_ERROR, { transactionHash });
+    }
+
+    const nonce = await this.getEvmTransactionCount(tx.from, tx.blockHash);
+    const extrinsic = await this._getExtrinsicByHashAtBlock(transactionHash, tx.blockHash);
+
+    const { args } = JSON.parse(extrinsic! as any).method;
```
Comment (xlc) [inline eth-providers/src/base-provider.ts:803]: better to check not undefined and return a real error or throw assertion error indicating it should never happen.
why `JSON.parse` is needed? `extrinsic` should be `GenericExtrinsic`?

Comment (shunjizhan) [reply eth-providers/src/base-provider.ts:803]: without JSON.pase, extrinsic looks something like 
```
Type {
  registry: TypeRegistry { createdAtHash: undefined },
  createdAtHash: undefined,
  _raw: Type(2) [Map] {
    'signature' => Type(5) [Map] {
      'signer' => [Type],
      'signature' => [Type],
      'era' => [Type],
      'nonce' => [Compact],
      'tip' => [Compact],
      registry: [TypeRegistry],
      createdAtHash: undefined
    },
    'method' => Type(2) [Map] {
      'callIndex' => [GenericCallIndex [Uint8Array]],
      'args' => [Struct [Map]],
      registry: [TypeRegistry],
      createdAtHash: undefined,
      _meta: [Type [Map]]
    },
    registry: TypeRegistry { createdAtHash: undefined },
    createdAtHash: undefined
  }
}
```
and after JSON.parse it looks like 
```
{
  signature: {
    signer: { id: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY' },
    signature: {
      sr25519: '0xae8968c6bfd626d0745c4bc82ca427f47aefe710b58aa23a2bdaa32607af406096388b12d4548badff48fc18e2f89c5de45361a794c99cd15684b46567236c8e'
    },
    era: { mortalEra: '0x3502' },
    nonce: 5,
    tip: 0
  },
  method: {
    callIndex: '0xb401',
    args: {
      target: '0x0230135fded668a3f7894966b14f42e65da322e4',
      input: '0x3d8d962000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000e8d4a510000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000010000010000000000000000000000000000000000000000000000000000000001000000',
      value: 1000000000000,
      gas_limit: 2000001,
      storage_limit: 1
    }
  }
}
```

on previous line we checked `if (!tx) { throw error }`, so I was assuming now tx exist, but let's be more safe the check extrinsic exist, I will add the check

Comment (shunjizhan) [reply eth-providers/src/base-provider.ts]: - added check to make sure extrinsic exist
- to native `toJSON` instead of `JSON.parse()`

**Review by ntduan — APPROVED** [review]

**Comment by gluneau** [comment]
I've git pulled, rush updated & builted
This is what I get.  Do you need anything else to troubleshoot this?

```
[1637028289672] DEBUG (1360574 on XPS-15-7590): incoming request
    jsonrpc: "2.0"
    method: "eth_sendRawTransaction"
    params: [
      "0xf...efdb9805"
    ]
    id: 10
[1637028289701] DEBUG (1360574 on XPS-15-7590): request completed
    id: 10
    jsonrpc: "2.0"
    result: "0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce"
[1637028289703] DEBUG (1360574 on XPS-15-7590): incoming request
    jsonrpc: "2.0"
    method: "eth_chainId"
    params: []
    id: 11
[1637028289703] DEBUG (1360574 on XPS-15-7590): request completed
    id: 11
    jsonrpc: "2.0"
    result: "0x253"
[1637028289704] DEBUG (1360574 on XPS-15-7590): incoming request
    jsonrpc: "2.0"
    method: "eth_getTransactionByHash"
    params: [
      "0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce"
    ]
    id: 12
[1637028289715] ERROR (1360574 on XPS-15-7590): request error
    methodName: "eth_getTransactionByHash"
    params: [
      "0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce"
    ]
    err: {
      "type": "Error",
      "message": "transaction hash not found (transactionHash=\"0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce\", code=UNKNOWN_ERROR, version=bodhi.js/providers/2.1.1-beta)",
      "stack":
          Error: transaction hash not found (transactionHash="0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce", code=UNKNOWN_ERROR, version=bodhi.js/providers/2.1.1-beta)
              at Logger.makeError (/home/bodhi.js/common/temp/node_modules/.pnpm/@ethersproject+logger@5.4.1/node_modules/@ethersproject/logger/src.ts/index.ts:225:28)
              at Logger.throwError (/home/bodhi.js/common/temp/node_modules/.pnpm/@ethersproject+logger@5.4.1/node_modules/@ethersproject/logger/src.ts/index.ts:237:20)
              at EvmRpcProvider.BaseProvider.getTransactionByHash (/home/bodhi.js/eth-providers/lib/base-provider.js:470:39)
              at processTicksAndRejections (internal/process/task_queues.js:95:5)
              at async Eip1193BridgeImpl.eth_getTransactionByHash (/home/bodhi.js/eth-rpc-adapter/src/eip1193-bridge.ts:220:29)
              at async Router.call (/home/bodhi.js/eth-rpc-adapter/src/router.ts:16:24)
              at async HTTPServerTransport.routerHandler (/home/bodhi.js/eth-rpc-adapter/src/transports/server-transport.ts:60:13)
              at async HTTPServerTransport.httpRouterHandler (/home/bodhi.js/eth-rpc-adapter/src/transports/http.ts:58:16)
      "reason": "transaction hash not found",
      "code": "UNKNOWN_ERROR",
      "transactionHash": "0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce"
    }
[1637028289715] DEBUG (1360574 on XPS-15-7590): request completed
    id: 12
    jsonrpc: "2.0"
    error: {
      "code": 6969,
      "message": "Error: transaction hash not found (transactionHash=\"0xa96f9805d78044c59dcfd64bd889b6fc46cbd051ab808fda94818a5215bc8bce\", code=UNKNOWN_ERROR, version=bodhi.js/providers/2.1.1-beta)"
    }
```

**Pull request merged**